servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpRequestDecoder.java

```diff
@@ -79,17 +91,15 @@ protected HttpRequestMetaData createMessage(final ByteBuf buffer,
                                                 final int firstStart, final int firstLength,
                                                 final int secondStart, final int secondLength,
                                                 final int thirdStart, final int thirdLength) {
-        if (thirdLength < 0) {
-            splitInitialLineError();
-        }
-
         return newRequestMetaData(nettyBufferToHttpVersion(buffer, thirdStart, thirdLength),
-                decodeHttpMethod(buffer.toString(firstStart, firstLength, US_ASCII)),
+                decodeHttpMethod(buffer, firstStart, firstLength),
                 buffer.toString(secondStart, secondLength, US_ASCII),
                 headersFactory().newHeaders());
     }
 
-    private static HttpRequestMethod decodeHttpMethod(final String methodName) {
+    private static HttpRequestMethod decodeHttpMethod(final ByteBuf buffer, final int start, final int length) {
+        buffer.forEachByte(start, length, ENSURE_UPPER_CASE);
```

Comment:
consider doing validation only for custom header names, most of the time we shouldn't have to do the validation because it is handled in the lookup (e.g. if it matches a cached entry it is upper case). It is worth paying the string conversion costs assuming most of the time we will get a valid string.